server/neptune/gateway/event/project_config_builder.go

```diff
@@ -0,0 +1,110 @@
+package event
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/logging"
+)
+
+// repoFetcher manages a cloned repo's workspace on disk for running commands.
+type repoFetcher interface {
```

Comment:
should we abstract out generating dir path in the fetcher and return a cleanup function: 
```
repoDir, cleanupFn, err := b.RepoFetcher.Fetch(event.Repo, event.Sha)
defer cleanupFn()
```
What do you think?